packages/server/src/tests/epics/epicSelection.test.ts

```diff
@@ -856,3 +857,113 @@ describe('deviceTypeMatchesFilter', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'SignedIn',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: true };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(true);
+    });
+
+    it('should fail if the test is requiring a user to be signed in and they are signed out', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'SignedIn',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: false };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(false);
+    });
+
+    it('should pass if the test is requiring a user to be signed out and they are signed out', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'SignedOut',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: false };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(true);
+    });
+
+    it('should fail if the test is requiring a user to be signed out and they are signed in', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'SignedOut',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: true };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(false);
+    });
+
+    it('should pass if the test is requiring a user to be either signed in or signed out and they are signed in', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'All',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: true };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(true);
+    });
+
+    it('should pass if the test is requiring a user to be either signed in or signed out and they are signed out', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'All',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: false };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(true);
+    });
+
+    it('should fail if the test is requiring a user to be signed in and isSignedIn returns undefined', () => {
+        const test: EpicTest = {
+            ...testDefault,
+            signedInStatus: 'SignedIn',
+        };
+
+        const targeting = { ...targetingDefault, isSignedIn: undefined };
+        const got = correctSignedInStatusFilter.test(test, targeting);
+
+        expect(got).toBe(false);
+    });
+
+    it('should pass if the test is requiring a user to be signed out and isSignedIn returns undefined', () => {
```

Comment:
Is this the desired behaviour?

Reply:
yes I think that's correct. In reality clients should always send `isSignedIn` now but it's good to handle this case